build/RuntimeStore.targets

```diff
@@ -0,0 +1,176 @@
+      <RuntimeStoreSymbolFiles Include="$(_RuntimeStoreOutputDirectory)symbols\**\*" />
+      <DepsFiles Include="$(_DepsOutputDirectory)**\*" />
+      <ExistingStoreManifests Include="$(_ExistingManifestsDirectory)*.xml" />
+    </ItemGroup>
+
+    <!-- Compose new store and symbols -->
+    <RepoTasks.ComposeNewStore
+      ExistingManifests="@(ExistingStoreManifests)"
+      NewManifests="@(RuntimeStoreManifestFiles)"
+      RuntimeStoreFiles="@(RuntimeStoreFiles)"
+      RuntimeStoreSymbolFiles="@(RuntimeStoreSymbolFiles)"
+      ManifestDestination="$(ArtifactsDir)aspnetcore-store-$(VersionPrefix)-$(VersionSuffix)-$(RID).xml"
+      StoreDestination="$(_StoreZipDirectory)store\"
+      SymbolsDestination="$(_SymbolsZipDirectory)"/>
+
+    <!-- Insert deps files -->
+    <Copy SourceFiles="@(DepsFiles)" DestinationFolder="$(_StoreZipDirectory)additionalDeps\%(RecursiveDir)" />
+
+    <ItemGroup>
+      <OutputZipFiles Include="$(_StoreZipDirectory)**\*" />
+      <OutputSymbolZipFiles Include="$(_SymbolsZipDirectory)**\*" />
+    </ItemGroup>
+
+    <!-- Create zips -->
+    <ZipArchive File="$(OutputZip)" SourceFiles="@(OutputZipFiles)" WorkingDirectory="$(_StoreZipDirectory)" Overwrite="true"/>
+    <ZipArchive File="$(OutputSymbolZip)" SourceFiles="@(OutputSymbolZipFiles)" WorkingDirectory="$(_SymbolsZipDirectory)" Overwrite="true" Condition="'@(OutputSymbolZipFiles)' != ''"/>
+  </Target>
+
+  <Target Name="CreateCommonManifest" DependsOnTargets="_PrepareRuntimeStoreBuildAssets">
+    <PropertyGroup>
+      <CommonManifestFileName>aspnetcore-store-$(VersionPrefix)-$(VersionSuffix)-common.xml</CommonManifestFileName>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackagesToTrim Include="runtime.win-arm64.runtime.native.system.data.sqlclient.sni" />
+    </ItemGroup>
+
+    <MSBuild Projects="$(_WorkRoot)RS.Manifest.csproj" Targets="Restore;GetPackageDefinitions" >
+      <Output TaskParameter="TargetOutputs" ItemName="_PackageDefinitions" />
+    </MSBuild>
+
+    <RepoTasks.CreateCommonManifest DestinationFilePath="$(ArtifactsDir)$(CommonManifestFileName)" PackageDefinitions="@(_PackageDefinitions)" Packages="@(PackagesToTrim)"/>
+  </Target>
+
+  <Target Name="AddManifestsToMetapackage">
+    <Error Text="COHERENCE_DROP_LOCATION is not specified. This requires a path like \\aspnetci\drops\Coherence\{branch}\{build-number}"
+        Condition=" '$(COHERENCE_DROP_LOCATION)' == '' " />
+
+    <ItemGroup>
+      <MetaPackageNupkg Include="$(COHERENCE_DROP_LOCATION)\ship\Microsoft.AspNetCore.All.*.nupkg" />
+      <ManifestFiles Include="$(ArtifactsDir)*.xml"/>
+    </ItemGroup>
+
+    <RemoveDir Directories="@(MetaPackageNupkg->'$(_WorkRoot)%(Filename)')" />
+    <Copy SourceFiles="@(MetaPackageNupkg)" DestinationFolder="$(ArtifactsDir)" />
+    <UnzipArchive File="$(ArtifactsDir)%(MetaPackageNupkg.FileName)%(MetaPackageNupkg.Extension)" Destination="@(MetaPackageNupkg->'$(_WorkRoot)%(Filename)')" />
```

Comment:
Instead of unzip/rezipping the nupkg, can we defer the nupkg creation until available files are ready?

Reply:
Let me give this a shot. I think it should work.

Reply:
Discussed offline, not worth it to pursue this.